File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    if s[0] == ')':
        return False
    stack = []
    for i in range(len(s)):
        if i == 0: 
            stack.append(s[i])
            continue
        if s[i] == '(':
            stack.append(s[i])
        elif stack and s[i] == ')':
            stack.pop()
            
    if stack:
        return False
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <iostream>
#include <vector>
using namespace std;

bool solution(string s)
{
    bool answer = true;
    vector<int> v;
    
    for (int i = 0; i < s.size(); i++){
        if (s[i] == '('){
            v.push_back('(');
        }
        else if (s[i] == ')'){
            if (v.empty())
                answer = false;
            else
                v.pop_back();
        }
    }
    
    if (not v.empty())
        answer = false;
        
    return answer;
}